### PR TITLE
chore: accept tree_config in new_with_params

### PR DESCRIPTION
**Pull request by richard-ramos**
go-waku embeds the zkey, vk and circuit in the binary, so it's not necessary to rely on external files when deploying go-waku service node. It uses `new_with_params` to be able to pass these parameters. 
However, this function does not allow passing the merkle tree configuration. 
This PR adds a new parameter `tree_config`  which would allow passing this configuration. 

In a future PR, rln-wasm should be updated to receive a JSON (not a string containing a json) in the `wasm_new` function

### CODE REVIEW

File: rln/src/public.rs [hunk 1]
```diff
@@ -134,18 +135,60 @@ impl RLN<'_> {
...
+            proving_key,
+            verification_key,
+            tree,
+        })
+    }
+
+    #[cfg(target_arch = "wasm32")]
     pub fn new_with_params(
         tree_height: usize,
-        #[cfg(not(target_arch = "wasm32"))] circom_vec: Vec<u8>,
```
Comment (rymnc) [inline rln/src/public.rs:148]: why is this line removed?

File: rln/src/public.rs [hunk 2]
```diff
@@ -159,12 +202,9 @@ impl RLN<'_> {
         let tree = PoseidonTree::default(tree_height)?;
 
         Ok(RLN {
-            #[cfg(not(target_arch = "wasm32"))]
-            witness_calculator,
             proving_key,
             verification_key,
             tree,
-            #[cfg(target_arch = "wasm32")]
```
Comment (rymnc) [inline rln/src/public.rs:167]: why is this line removed?

**Review by rymnc — COMMENTED** [review]
LGTM in general, but left a couple comments! thanks for this pr!

File: rln/src/public.rs (hunk 1 above)
Comment (richard-ramos) [reply rln/src/public.rs:148]: The reason why some lines are removed is that there are now 2 `new_with_params` functions (in line 151 and 190). One of these drops the code/params that are not needed when compiling to wasm, and another one which is used for all other architectures.

I tried to keep a single new_with_params function but the code ended up being harder to read, so I ended up creating two, which simplifies reading the code with the tradeoff of adding (some) duplication of code

**Review by rymnc — APPROVED** [review]
LGTM, thanks for this!
